aldryn_forms/admin/formdata.py

```diff
@@ -20,5 +30,89 @@ class FormDataAdmin(BaseFormSubmissionAdmin):
+
+                # A user can add fields to the form over time,
+                # knowing this we use the latest form submission as a way
+                # to get the latest form state, but this means that if a field
+                # was removed then it will be ignored :(
+                first_entry = entries.order_by('-sent_at')[0]
+
+                # what follows is a bit of dark magic...
+                # we call the export view in tablib with a headers dictionary, this dictionary
+                # maps a key to a callable that gets passed a form submission instance and returns the value
+                # for the field, we have to use a factory function in order to avoid a closure
+
+                def clean_data(label, position):
+                    def _clean_data(obj):
+                        field = None
+                        value = ''
+
+                        try:
+                            field = obj.get_form_data()[position]
+                        except IndexError:
+                            pass
+
+                        if field and field.label == label:
+                            # sanity check
+                            # we need this to make sure that the field label and position remain constant
+                            # otherwise we'll confuse users if a field was moved.
+                            value = field.value
+                        return value
+                    return _clean_data
+
+                fields = first_entry.get_form_data()
+
+                # used to keep track of occurrences
+                # in case a field with the same name appears multiple times in the form.
+                occurrences = defaultdict(lambda: 1)
+                headers = SortedDict()
+
+                for position, field in enumerate(fields):
+                    label = field.label
+
+                    if label in headers:
+                        occurrences[label] += 1
+                        label = u'%s %s' % (label, occurrences[label])
+                    headers[label] = clean_data(field.label, position)
+
+                headers[ugettext('Language')] = 'language'
+                headers[ugettext('Submitted on')] = 'sent_at'
+
+                do_export = partial(
+                    export,
+                    request=request,
+                    queryset=entries,
+                    model=entries.model,
+                    headers=headers,
+                    filename=filename
+                )
+
+                try:
+                    # Since django-tablib 3.1 the parameter is called file_type
+                    response = do_export(file_type='xls')
+                except TypeError:
+                    response = do_export(format='xls')
+                return response
+            else:
+                self.message_user(request, _("No records found"), level=messages.WARNING)
+                export_url = 'admin:{}'.format(self.get_admin_url('export'))
```

Comment:
if it should work on python 2.6 please use indexes for string formatting like ``{0}``.